source/patterns/@aws-solutions-constructs/core/lib/s3-bucket-helper.ts

```diff
@@ -31,13 +30,19 @@ export interface BuildS3BucketProps {
    * @default - Default props are used
    */
   readonly bucketProps?: s3.BucketProps
+  /**
+   * User provided props to override the default props for the S3 Logging Bucket.
+   *
+   * @default - Default props are used
+   */
+  readonly loggingBucketProps?: s3.BucketProps
 }
 
 export function buildS3Bucket(scope: Construct, props: BuildS3BucketProps, bucketId?: string): [s3.Bucket, s3.Bucket?] {
   if (props.bucketProps) {
-    return s3BucketWithLogging(scope, props.bucketProps, bucketId);
+    return s3BucketWithLogging(scope, props.bucketProps, bucketId, props.loggingBucketProps);
   } else {
-    return s3BucketWithLogging(scope, DefaultS3Props(), bucketId);
+    return s3BucketWithLogging(scope, DefaultS3Props(), bucketId, props.loggingBucketProps);
```

Comment:
Would this be clearer written

`
return s3BucketWithLogging( scope,
                                                   props.bucketProps || DefaultS3Props(),
                                                   bucketId,
                                                   props.loggingBucketProps);
 `

Or...do we even need this condition? There is code in lines 153-158 of S3BucketWithLogging that does this check again. Wouldn't all this still work if we just had this?

`
return s3BucketWithLogging( scope,
                                                   props.bucketProps,
                                                   bucketId,
                                                   props.loggingBucketProps);
 `

Reply:
That actually becomes a bigger question of do we need this function at all, or could all of s3BucketWithLogging be moved here?

Reply:
buildS3Bucket is called in multiple place so I left it as is